File: c++/Two_Sum.cpp
#include <vector>
#include<stdint.h>
#include<iostream>
#include<unordered_map>
#include<math.h>
using namespace std;
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> map;
        for(int i=0; i<nums.size();i++){
            int complement = target - nums[i];
            if(map.find(complement) != map.end()){
                return {map[complement], i};
            }
            map[nums[i]]=i;
        }
        return {};
    }
};

int main(){
    Solution s;
    vector<int> val;
    vector<int> nums;
    nums={1,3,4,45,7,8};
    int target;
    target = 15;
    val=s.twoSum(nums, target);
    for(int i=0;i<val.size();i++){
        cout<<val[i]<<endl;
    }
    return 0;
}

File: java/Two_Sum.java
import java.util.*;

class Solution {
    public int[] Two_Sum(int[] nums, int target){
        Map<Integer, Integer> map =new HashMap<>();
        for(int i=0;i<nums.length;i++){
            int complement;
            complement = target - nums[i];
            if(map.containsKey(complement)){
                return new int[] {map.get(complement), i };
            }
            map.put(nums[i], i);
        }
        return new int[] {};
     }
}

public class Two_Sum {
    public static void main(String[] args){
        Solution s = new Solution();
        int[] nums = {1,3,4,45,7,8};
        int target = 15;
        int[] result =s.Two_Sum(nums, target);
        for(int val : result){
            System.out.println(val);
        }
    }   
}